app/controllers/application_controller.rb

```diff
@@ -43,4 +45,13 @@ def destroy_user_if_current_user
     @user = User.new
   end 
   
+  private
+  def extract_locale_from_accept_language_header
```

Comment:
* Only return `locale` if `locale.in?(User::LANGUAGES)`

* Cover this method with a spec. Does it correctly derive `en` from `en`, `en-US`, etc?

* For easier testability, you can extract this method to a new dedicated class like `app/lib/http_accept_lang_parser.rb`. That class would accept a string as an argument, passed from ApplicationController:

```
def extract_locale_from_accept_language_header
  HttpAcceptLangParser.parse(request.env['HTTP_ACCEPT_LANGUAGE'])
end
```

That way in your unit tests you only test HttpAcceptLangParser and don't have to worry about instantiating ApplicationController or manipulating `request`